src/app/clusters/ota-requestor/OTARequestor.h

```diff
@@ -22,16 +22,30 @@
 
 #pragma once
 
+#include <app/CASESessionManager.h>
+#include <protocols/bdx/BdxMessages.h>
+
 #include "BDXDownloader.h"
 #include "OTARequestorDriver.h"
 #include "OTARequestorInterface.h"
-#include <app/CASESessionManager.h>
-#include <protocols/bdx/BdxMessages.h>
+
+namespace chip {
+
+using namespace app::Clusters::OtaSoftwareUpdateProvider::Commands;
 
 // This class implements all of the core logic of the OTA Requestor
 class OTARequestor : public OTARequestorInterface
```

Comment:
By now, the interface does nothing. Suggest removing the interface, since the way OTARequestor is being written, there is no clear separation of platform, cluster client, etc, that would justify an additional OTARequestorInterface. In other words, OTARequestorInterface currently adds nothing since this is a full monolothic implementation. OTARequestorInterface is not used direclty anywhere

Reply:
The OTARequestorInterface is necessary for allowing a full custom implementation of the OTA Requestor. If for some reason the SDK implementation is not suitable to some vendor this would be the only way to hook up a different implementation.

Reply:
But the `OTARequestorInterface` has very few methods, and the only time it's used is in handling AnnounceOTAServer. Otherwise, it's not actually used at all.

If it was truly an interface that allowed different implementations, it would have a real abstract public API that accounts for most operations, including TriggerImmediateQuery, etc.

Reply:
Yes, handler for AnnounceOTAServer is the only compile-time entry into the Requestor from the Matter stack. Everything else is dynamic/runtime (like command response callbacks). 

If a platform chooses to use its custom Requestor implementation over the SDK-provided one it is up to the platform to define/use/not use an abstract API for the rest of the operation; this stops being SDK's responsibility.